lib/src/repository/devices/devices_repository.dart

```diff
@@ -246,32 +256,37 @@ class SupportsDiscreteOnOff extends Property<bool> {
   SupportsDiscreteOnOff(bool value) : super('supportsDiscreteOnOff', value);
 }
 
+class AvailableFanMode extends Property<String> {
+  AvailableFanMode(String value) : super('availableFanModes', value);
+}
+
 class SupportsIsJammed extends Property<bool> {
   SupportsIsJammed(bool value) : super('supportsIsJammed', value);
 }
 
 class LockTrait extends Trait {
-  LockTrait(State state, List<Property> properties)
-      : super('lock', state, properties);
+  LockTrait(State state, Set<Property> properties)
+      : super('lock', {state}, properties);
 }
 
 class PowerTrait extends Trait {
-  PowerTrait(State state, List<Property> properties)
-      : super('power', state, properties);
+  PowerTrait(State state, Set<Property> properties)
+      : super('power', {state}, properties);
 }
 
 class ThermostatTrait extends Trait {
   ThermostatTrait(
-    State state,
-  ) : super('thermostat_setting', state, []);
+    Set<State> states,
+    Set<Property> properties,
```

Comment:
Not sure if this is what we want to do. There is a good argument that each trait should have fields for the properties as it seems they will always exist and only pertain to specific traits.